packages/blockchain-extension/test/commands/restartFabricRuntime.test.ts

```diff
@@ -238,9 +240,8 @@ describe('restartFabricRuntime', () => {
 
         executeCommandSpy.should.not.have.been.calledWith(ExtensionCommands.DISCONNECT_GATEWAY);
         executeCommandSpy.should.not.have.been.calledWith(ExtensionCommands.DISCONNECT_ENVIRONMENT);
-
-        logSpy.getCall(0).should.have.been.calledWithExactly(LogType.INFO, undefined, 'restartFabricRuntime');
-        logSpy.getCall(1).should.have.been.calledWithExactly(LogType.ERROR, `Failed to restart ${environment.getName()}: ${error.message}`, `Failed to restart ${environment.getName()}: ${error.toString()}`);
+        popupStub.should.have.been.calledWithExactly(`Failed to restart ${environment.getName()}: ${error.message}`,
+                    `Failed to restart ${environment.getName()}: ${error.toString()}`);
```

Comment:
This is more of a test structure comment. I've noticed in our team we usually do the following:
 - any prep/create stubs, etc
 - run the command we are interested in testing
 - check the stubs we created for the required functions
 - check the stubs for running/executing other commands
 - check the spies.

Although not necessarily every single tests will be written in this way, I find that most are.
More importantly though, if the other tests in that file follow a structure like this, if you create your tests or add info to existing tests using the same structure, it will make them much easier to read and check in the future, because we'll 'know' where to look for the info.

So the couple of places where you check your popupStub after the spies, they should be in fact after executing the command of interest, and before checking other command executions and spies.

In the tests where you create spies, leaving them at the bottom of the test like you did is the best option.